test/server/cards/02_SHD/upgrades/ArmedToTheTeeth.spec.ts

```diff
@@ -0,0 +1,35 @@
+describe('Armed to the teeth upgrade', function () {
+    integration(function (contextRef) {
+        describe('Armed to the teeth\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [{ card: 'bokatan-kryze#fighting-for-mandalore', upgrades: ['armed-to-the-teeth'] }],
+                        spaceArena: ['collections-starhopper'],
+                        base: { card: 'echo-base', damage: 10 }
+                    },
+                    player2: {
+                        groundArena: ['battlefield-marine'],
+                        base: { card: 'echo-base', damage: 0 }
+                    }
+                });
+            });
+
+            it('ground unit has 5 power and on attack gives other unit +2, which removes at end of phase', function () {
+                const { context } = contextRef;
+                expect(context.bokatanKryze.getPower()).toBe(5);
+                expect(context.collectionsStarhopper.getPower()).toBe(2);
+
+                context.player1.clickCard(context.bokatanKryze);
+                context.player1.clickCard(context.p2Base);
+                context.player1.clickCard(context.collectionsStarhopper);
```

Comment:
Any time we write a card with a `targetResolver` we always need to have at least one instance of `toBeAbleToSelectExactly` to make sure the filters are set up right